### ISSUE TITLE
Upgrade trouble with many printers

### ISSUE THREAD

**Issue by severach** [CauldronDevelopmentLLC/oh-brother#16]
HL-5250DN: pysnmp.proto.errind.RequestTimedOut: No SNMP response received before timeout
HL-L2340D: Firmware already up to date
MFC-7360N: Firmware already up to date
MFC-J5620DW: No firmware update info path found
MFC-J6710DW: pysnmp.proto.errind.RequestTimedOut: No SNMP response received before timeout
MFC-J870DW: No firmware update info path found
MFC-J875DW: No firmware update info path found
MFC-L2740DW: Upgrade sent but didn't apply

More
HL-2070N: No SNMP response received before timeout
HL-2270DW:  serial = None; model = None; spec = None

**Comment by jcoffland** [comment]
Duplicate of #6.

**Comment by mwaterbu** [comment]
@jcoffland I'm not sure this issue is a duplicate.
I also noticed the same behavior with my HL-2270DW (everything was showing none), but I was able to fix it by making a couple changes to the string processing of the SNMP returned values (in my fork: b28457829d9a5e15534ec70971df5a9ae0cbadd4). Unfortunately, I don't have any other printers to test this code with, to see the best way to merge this functionality and add support for at least this printer.
